internal/proxy/clientgo_cached_proxy_resolver.go

```diff
@@ -155,6 +155,17 @@ func (p *clientgoCachedProxyResolver) ObjectExists(obj client.Object) (bool, err
 	return exists, err
 }
 
+func (p *clientgoCachedProxyResolver) IsReady() bool {
+	// If the proxy is has no database, it is only ready after a successful sync
+	// Otherwise, it has no configuration loaded
+	if p.dbmode == "off" {
+		return len(p.lastConfigSHA) > 0
```

Comment:
It should only break if it's removed entirely (in which case the invalid reference on vet will call out the issue), correct? We don't really care what the contents are or how we generate them, only whether it's initialized or not.

Theoretically you could initialize it to a static value, but that'd be quite odd given what we use it for.

Reply:
Before this change, the responsibility of the `lastConfigSHA` field of proxy was the following:
- (opportunistically) provide a config fingerprint to de-duplicate subsequent proxy updates

After this change, the responsibility of  the `lastConfigSHA` field of proxy is the following:
- (opportunistically) provide a config fingerprint to de-duplicate subsequent proxy updates
- be a critical part of the readiness mechanism

I agree that lastConfigSHA will very likely work successfully in this capacity, but it's not a signal that was designed for this purpose.

Here's a small tongue-in-cheek parabola to show where I'm coming from: think cars - suppose that the airbag system needs a signal whether the car is in use or not.
Here are some example signals that would probably be robust enough to do well and not cause any issues in safety statistics:
- whether the dashboard is backlit
- whether there is pressure in the brake system
- whether the RPM-meter is showing a value between 1 and 10000

Any change to these systems may leave the implicit dependency of the airbag system on backlight/brakes/gauges unnoticed, and cause the depending system to malfunction.

Back to readiness checks - I believe in a good system design practice that signals should be used to the extent of their intended purpose. In this case, we're clearly going beyond the intended purpose of the lastConfigSHA, and we have two options available:
1. extend the responsibility of lastConfigSHA as explained above, and document it clearly
2. create a new signal with obvious semantics

I'm advocating for (2) because I believe it provides for better readability and maintainability, today we're at (1) minus the documentation.

Reply:
Were these unrelated I'd agree, but readiness here isn't an abstract concept or a combination of multiple conditions--we care specifically about whether we've applied configuration, so the car analogy version might be more like "if RPM > 4000, shift into a higher gear" (for a vastly oversimplified automatic transmission).

Separate value works better with locking though, since we can avoid touching it each time and blocking the channel handler if there happens to be a readiness check in process.